src/lib.rs

```diff
@@ -5703,3 +5892,213 @@ mod tests {
         }
     }
 }
+
+#[cfg(kani)]
+mod proofs {
+    use super::*;
+
+    impl kani::Arbitrary for DstLayout {
+        fn any() -> Self {
+            let align: NonZeroUsize = kani::any();
+            let size_info: SizeInfo = kani::any();
+
+            kani::assume(align.is_power_of_two());
+            kani::assume(align < DstLayout::MAX_ALIGN);
+
+            // For testing purposes, we most care about instantiations of
+            // `DstLayout` that can correspond to actual Rust types. We use
+            // `Layout` to verify that our `DstLayout` satisfies the validity
+            // conditions of Rust layouts.
+            kani::assume(
+                match size_info {
+                    SizeInfo::Sized { _size } => Layout::from_size_align(_size, align.get()),
+                    SizeInfo::SliceDst(TrailingSliceLayout { _offset, _elem_size }) => {
+                        // `SliceDst`` cannot encode an exact size, but we know
+                        // it is at least `_offset` bytes.
+                        Layout::from_size_align(_offset, align.get())
```

Comment:
This is too strict - it's valid to have types whose trailing field offset does not satisfy the overall type's alignment. E.g.:

```rust
#[repr(C)]
struct Foo {
    a: u64,   // at byte offset 0
    b: u8,    // at byte offset 8
    c: [u16], // at byte offset 10
}
```

This type has alignment 8, but the trailing slice offset is 10, which is not a multiple of 8.

Reply:
`Layout::from_size_align` does not enforce that size satisfies alignment, since it's intended to be used for representing fragments of layouts.